From a3491db018faf7255606d04cdc3477c9db4faca1 Mon Sep 17 00:00:00 2001
From: Frank Li <Frank.Li@nxp.com>
Date: Mon, 7 May 2018 17:24:32 -0500
Subject: [PATCH 3740/5242] MLK-18237-2 XRDC: DSP: add mu power resource to
 avoid crash

commit  8d45acb78393bf099b9794bca7707c552f2b2cf5 from
https://source.codeaurora.org/external/imx/linux-imx.git

[    2.300213] Unhandled fault: synchronous external abort (0x96000210) at 0xffff000014dd0000
[    2.308584] Internal error: : 96000210 [#1] PREEMPT SMP
[    2.313813] Modules linked in:
[    2.316875] CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.9.88-04903-ga209cd8 #464
[    2.324271] Hardware name: Freescale i.MX8QXP MEK (DT)
[    2.329407] task: ffff80083a088000 task.stack: ffff80083a034000
[    2.335329] PC is at MU_Init+0x0/0x38
[    2.338994] LR is at dsp_mu_init+0xb8/0x140

Signed-off-by: Frank Li <Frank.Li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index d47669a..3bd59869 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -556,11 +556,20 @@
 				};
 			};
 
-			pd_dsp: PD_AUD_DSP {
-				reg = <SC_R_DSP>;
+			pd_dsp_mu: PD_DSP_MU {
+				reg = <SC_R_MU_13A>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_audio>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_dsp: PD_AUD_DSP {
+					reg = <SC_R_DSP>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_audio>;
+				};
 			};
+
 			pd_dsp_ram: PD_AUD_OCRAM {
 				reg = <SC_R_DSP_RAM>;
 				#power-domain-cells = <0>;
-- 
1.7.9.5

